dandelion-voting/contracts/DandelionVoting.sol

```diff
@@ -232,23 +246,29 @@ contract DandelionVoting is IForwarder, IACLOracle, AragonApp {
     * @notice Returns whether the sender has voted on the most recent open vote or closed unexecuted vote.
     * @dev IACLOracle interface conformance. The ACLOracle permissioned function should specify the sender
     *      with 'authP(SOME_ACL_ROLE, arr(sender))', where sender is typically set to 'msg.sender'.
-    * @param _who The address to check if can perform (ignored if `_how` contains an address)
     * @param _how Array passed by Kernel when using 'authP()'. First item should be the address to check can perform.
     * return False if the sender has voted on the most recent open vote or closed unexecuted vote, true if they haven't.
     */
-    function canPerform(address _who, address, bytes32, uint256[] _how) external view returns (bool) {
+    function canPerform(address, address, bytes32, uint256[] _how) external view returns (bool) {
         if (votesLength == 0) {
             return true;
         }
 
-        Vote storage latestVote = votes[votesLength];
-        address sender = _how.length > 0 ? address(_how[0]) : _who;
+        require(_how.length > 0, ERROR_ORACLE_SENDER_MISSING);
+        require(_how[0] < 2**160, ERROR_ORACLE_SENDER_TOO_BIG);
+        require(_how[0] != 0, ERROR_ORACLE_SENDER_ZERO);
+        address sender = address(_how[0]);
+
+        uint256 senderLatestYeaVoteId = latestYeaVoteId[sender];
```

Comment:
It's a trade-off I guess. Depending on where it goes, it will likely make the function call cheaper for those who haven't voted yet but more expensive for those that have. 

I should also point out I've opted for readability over gas efficiency here, so to be consistent with my approach (eg creating bools and evaluating them all at once in the return statement as opposed to the `_canExecute()` function which evaluates throughout) it would make it more expensive for everyone. However, if you think we should optimise for gas cost I can have a go at that.

Reply:
Giving it a second thought, it's not going to be too much of a gas saving since in the context of Dandelion,  it's more likely for users that haven't voted, will exit the org since this function is being executed (or are redeeming a partial amount of their tokens but less likely).
So i think that the % of times that a voter had ever voted in this context should be minimal compared to the times it had, where `senderLatestYeaVoteId ` would always be greater than zero.

So yeah better go with the better readability.